### PR TITLE
Return null for non-numeric error code

### PR DESCRIPTION
**Pull request by daliwei**
@pierre As discussed offline, this PR will 1) return `null` if there isn't any numeric error code found; 2) format the error message except exceptions.

### CODE REVIEW

File: src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentTransactionInfoPlugin.java [hunk 1]
```diff
@@ -436,4 +412,13 @@ private static void computeLengthArray(final String className, final int[] dotAr
...
         return ImmutableList.<PluginProperty>copyOf(propertiesWithDefaults);
     }
+
+    @VisibleForTesting
+    static String formatErrorMessage(final String gatewayError) {
+        if(Strings.isNullOrEmpty(gatewayError)) {
+            return gatewayError;
+        }
+        //Remove any extra spaces between word and convert all to lower-case
+        return gatewayError.replaceAll("\\s+", " ").trim().toLowerCase();
```
Comment (pierre) [inline src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentTransactionInfoPlugin.java]: Looking at https://github.com/killbill/killbill-payment-retries-plugin/pull/1, we don't need this anymore, do we?

Comment (daliwei) [reply src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentTransactionInfoPlugin.java]: You mean the format part? Yes, that is not that necessary.

Comment (daliwei) [reply src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentTransactionInfoPlugin.java]: Removed in b2ac7f3

**Comment by pierre** [comment]
@daliwei Could you take a look at the [tests failures](https://travis-ci.org/killbill/killbill-adyen-plugin/jobs/271038985)?

**Pull request merged**